modules/hal_nordic/Kconfig

```diff
@@ -205,6 +205,17 @@ endif
 
 endmenu # NRF_802154_SER_HOST || NRF_802154_SER_RADIO
 
+if (NRF_802154_RADIO_DRIVER || NRF_802154_SERIALIZATION)
+
+config NRF_802154_CARRIER_FUNCTIONS
+	bool "Enable 802.15.4 carrier functions"
+	default n
```

Comment:
No need to explicitly set it: https://docs.zephyrproject.org/1.14.0/guides/kconfig/index.html#redundant-defaults.